### PR TITLE
[jmxfetch] adding knobs for statsd client queue size + telemetry

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?
This PR adds support for jmxfetch dogstatsd client queue size and telemetry knobs. These require https://github.com/DataDog/jmxfetch/pull/390 to be merged prior.
 

### Motivation
Some higher metric volume jmxfetch use-cases can benefit from large queue sizes to avoid metric drops. Additionally, adding telemetry is a massive aid in troubleshooting these kinds of issues as we can identify drops on the client side.

### Describe how to test/QA your changes
 We can use our internal systems to validate the fix.
 - Find a cluster with a high JMX metric volume. You can sync with @mathispesch to find a suitable cluster.
 - Enable the correct options with respect to the queue size and check there no metric drops/gaps on relevant metrics. You can also use the client telemetry to validate this.
 
 To enable the client telemetry, simply set the `jmx_statsd_telemetry_enabled` option to `true` in `/etc/datadog-agent/datadog.yaml` or with the `DD_JMX_STATSD_TELEMETRY_ENABLED` option. You should then be able to see incoming telemetry for the JMXFetch java statsd client.  

The bump to JMXFetch 0.46.0 here also introduces two other small-impact changes:
 - https://github.com/DataDog/jmxfetch/pull/375
 - https://github.com/DataDog/jmxfetch/pull/376
 
 Testing the latter is straight-forward, if JMXFetch starts with no issues, then all is well.
 For https://github.com/DataDog/jmxfetch/pull/375, you can try the following:
 
 1. Set a JMX config with multiple services defined in the init section:
 ```yaml
 init_config:
  service:
    - global
    - foo
    - bar
    - test

instances:
  # ...
 ```
 
 2. Verify that every time series submitted by the JMX instances is tagged with *all* the service tags. In-app you should be able to aggregate by individual services no problems if need be.

3. Also verify there are no issues with backward-compatible configs with a single service tag:
```yaml
init_config:
  service: foo

instances:
  # ...
```  

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by gh123man — APPROVED** [review]

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -1875,6 +1875,18 @@ api_key:
 #
 # jmx_reconnection_timeout: 60
 
+## @param jmx_statsd_client_queue_size - integer - optional - default: 4096
+## @env DD_JMX_STATSD_CLIENT_QUEUE_SIZE - integer - optional - default: 4096
+## Determines the size of the JMXFetch statds client queue size (in number of elements).
```
Comment (kayayarai) [inline pkg/config/config_template.yaml:1880]:
```suggestion
## Determines the size of the JMXFetch statds client queue (in number of elements).
```

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -1875,6 +1875,18 @@ api_key:
...
+#
+# jmx_statsd_client_queue_size: 4096
+
+## @param jmx_statsd_telemetry_enabled - boolean - optional - default: false
+## @env DD_JMX_STATSD_TELEMETRY_ENABLED - boolean - optional - default: false
+## Defines whether or not JMXFetch statsd client telemetry will enabled.
```
Comment (kayayarai) [inline pkg/config/config_template.yaml:1886]:
```suggestion
## Specifies whether the JMXFetch statsd client telemetry is enabled.
```

**Review by kayayarai — APPROVED** [review]
A couple edits, and 👍

File: release.json [hunk 3]
```diff
@@ -20,8 +20,8 @@
         "INTEGRATIONS_CORE_VERSION": "master",
         "OMNIBUS_SOFTWARE_VERSION": "master",
         "OMNIBUS_RUBY_VERSION": "datadog-5.5.0",
-        "JMXFETCH_VERSION": "0.44.6",
-        "JMXFETCH_HASH": "dbd31f982328a3cf6b44c65b06812a14daf24172a82ab6266be1d208e0f420e9",
+        "JMXFETCH_VERSION": "0.46.0",
+        "JMXFETCH_HASH": "<unknown>",
```
[the remark this answers is not in the document]
Comment (truthbk) [reply release.json:24]: I won't be merging until I update this. Thanks @gh123man

File: pkg/config/config_template.yaml [hunk 4]
```diff
@@ -1875,6 +1875,18 @@ api_key:
 #
 # jmx_reconnection_timeout: 60
 
+## @param jmx_statsd_client_queue_size - integer - optional - default: 4096
+## @env DD_JMX_STATSD_CLIENT_QUEUE_SIZE - integer - optional - default: 4096
+## Determines the size of the JMXFetch statds client queue (in number of elements).
+#
+# jmx_statsd_client_queue_size: 4096
```
Comment (olivielpeau) [inline pkg/config/config_template.yaml:1882]: This is a purely internal setting, I don't think we should document it in the config_template. Could you add this documentation as a comment on `pkg/config/config.go` instead? Also, could you mention what `elements` mean in this context? (does each element represent a metric/service check?)

File: releasenotes/notes/jmxfetch-statsd-qsize-telemetry-d12fe0746f750cc0.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    Addition of two knobs to configure JMXFetch stastsd client:
```
Comment (olivielpeau) [inline releasenotes/notes/jmxfetch-statsd-qsize-telemetry-d12fe0746f750cc0.yaml:11]:
```suggestion
    Addition of two knobs to configure JMXFetch statsd client:
```

File: releasenotes/notes/jmxfetch-statsd-qsize-telemetry-d12fe0746f750cc0.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
```
Comment (olivielpeau) [inline releasenotes/notes/jmxfetch-statsd-qsize-telemetry-d12fe0746f750cc0.yaml:9]:
```suggestion
enhancements:
```

**Review by olivielpeau — COMMENTED** [review]
Could you add to your QA plan the QA that's needed for the changes in JMXFetch 0.45?

File: pkg/config/config_template.yaml (hunk 4 above)
Comment (truthbk) [reply pkg/config/config_template.yaml:1882]: Sounds good.

**Review by olivielpeau — APPROVED** [review]

**Pull request merged**